File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int,int> mp;
        for(auto c:nums){
            mp[c]++;
        }
        auto cmp = [](const pair<int,int>& a, const pair<int,int>& b) {
            return a.second < b.second; 
        };
        priority_queue<pair<int,int>, vector<pair<int,int>>, decltype(cmp)> pq(mp.begin(), mp.end(), cmp);

        vector<int> ans;
        while(k--){
            ans.emplace_back(pq.top().first); pq.pop();
        }
        return ans;
    }
};

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-0.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> minHeap = new PriorityQueue<>((a,b) -> a-b);
        for(int c:nums){
            minHeap.add(c);
            if(minHeap.size() > k)
                minHeap.poll();
        }
        return minHeap.poll();
    }
}
